recommendations.py

```diff
@@ -5,278 +5,280 @@
-
-        if not final_recommendations:
-            final_recommendations.append(create_book_lists_helper("No books found!", []))
-
-        return results_in_json(final_recommendations)
-    except Exception as e:
-            print(f"Error in get_recommendations_by_book: {e}")
-
-def get_recommendations_by_author(author_name):
-    final_recommendations = []
-    final_recommendations.append(recommendation_by_given_author(author_name))
-    return results_in_json(final_recommendations)
-
-def get_recommendations_by_publisher(publisher_name):
-    final_recommendations = []
-    final_recommendations.append(recommendation_by_given_publisher(publisher_name))
-    return results_in_json(final_recommendations)
-
-def get_recommendations_by_year(year):
-    final_recommendations = []
-    final_recommendations.append(recommendations_by_year(year))
-    return results_in_json(final_recommendations)
-
-def get_recommendations_by_location(location):
-    final_recommendations = []
-    final_recommendations.append(recommendations_by_location(location))
-    return results_in_json(final_recommendations)
\ No newline at end of file
+                #no books from the same year
+                if (len(same_place_books) == 0):
+                    return self.create_book_lists_helper("oops! No recommendations for place input", books_list)
+                
+            places = ((df_recommendation_dataset['City'].str.lower() == same_place_books.iloc[0]['City'].lower()) |
+                    (df_recommendation_dataset['State'].str.lower() == same_place_books.iloc[0]['State'].lower()) |
+                    (df_recommendation_dataset['Country'].str.lower() == same_place_books.iloc[0]['Country'].lower()))
+        
+            if places.any():
+                same_place_books = df_recommendation_dataset[places]
+                #top 5 rated books
+                same_place_books = same_place_books.sort_values(by = "Book-Rating", ascending=False)[:5]
+                same_place_books = same_place_books.drop_duplicates(subset=["Book-Title"])
+                if(len(same_place_books) == 0):
+                    return self.create_book_lists_helper("oops! No recommendations for place input", books_list)    
+                for book_info in same_place_books.values.tolist():
+                    recommended_book = self.Book(book_info[0], book_info[4], book_info[1])
+                    books_list.append(recommended_book)
+                return self.create_book_lists_helper("Trending books at the same location", books_list)
+            else:
+                return self.create_book_lists_helper("oops! No recommendations for place input", books_list)    
+        except Exception as e:
+            print(f"Error in recommendation_by_same_place: {e}")
+
+    # Converting result to JSON format for frontend
+    def results_in_json(final_recommendations): 
```

Comment:
The "results_in_json" method is defined to take only one positional argument but is being called with two arguments, Modify the code to 'def results_in_json(self, final_recommendations).'